app/validators/uri_validator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class UriValidator < ActiveModel::EachValidator
+  DEFAULT_ACCEPTED_SCHEMES = /^https?$/
+  DEFAULT_OPTIONAL_PARTS = %i[port].freeze
+  DEFAULT_FORBIDDEN_PARTS = %i[userinfo registry path opaque query fragment].freeze
+
+  def validate_each(record, attribute, value)
+    uri = URI.parse(value)
+    raise URI::InvalidURIError if !valid_scheme?(uri.scheme) || uri.host.blank? || forbidden_part?(uri)
```

Comment:
I'd use another class than `InvalidURIError`
Probably not even raising an error here but only setting the error message to the record

Reply:
You are right. It shouldn't raise the exception anyway. I'll fix that.